src/app/tide/peptide_mods3.cc

```diff
@@ -61,275 +61,251 @@ class IModsOutputter {
-                   HeadedRecordWriter* final_writer,
-                   unsigned long long memory_limit)
-    : tempDir_(tmpDir), proteins_(proteins), modTable_(vmt),
-      maxMods_(0), writer_(final_writer), totalWritten_(0), 
-      temp_file_cnt_(0), memory_limit_(memory_limit) {
-    modMaxCounts_.clear();
-    const vector<int>* maxCounts = vmt->MaxCounts();
-    for (char c = 'A'; c <= 'Z'; c++) {
-      for (int i = 0; i < vmt->NumPoss(c, MOD_SPEC); i++) {
-        int deltIx = vmt->PossDeltIx(c, i, MOD_SPEC);
-        int maxCt = maxCounts->at(deltIx);
-        modMaxCounts_[deltIx] = maxCt;
-        maxMods_ += maxCt;
-      }
-      for (int i = 0; i < vmt->NumPoss(c, NTPEP); i++) {
-        int deltIx = vmt->PossDeltIx(c, i, NTPEP);
-        int maxCt = maxCounts->at(deltIx);
-        modMaxCounts_[deltIx] = maxCt;
-        maxMods_ += maxCt;
-      }
-      for (int i = 0; i < vmt->NumPoss(c, NTPRO); i++) {
-        int deltIx = vmt->PossDeltIx(c, i, NTPRO);
-        int maxCt = maxCounts->at(deltIx);
-        modMaxCounts_[deltIx] = maxCt;
-        maxMods_ += maxCt;
-      }
-      for (int i = 0; i < vmt->NumPoss(c, CTPRO); i++) {
-        int deltIx = vmt->PossDeltIx(c, i, CTPRO);
-        int maxCt = maxCounts->at(deltIx);
-        modMaxCounts_[deltIx] = maxCt;
-        maxMods_ += maxCt;
-      }
-      for (int i = 0; i < vmt->NumPoss(c, CTPEP); i++) {
-        int deltIx = vmt->PossDeltIx(c, i, CTPEP);
-        int maxCt = maxCounts->at(deltIx);
-        modMaxCounts_[deltIx] = maxCt;
-        maxMods_ += maxCt;
-      }
-    }
-    if (maxMods_ > FLAGS_max_mods) {
-      maxMods_ = FLAGS_max_mods;
+  ModsOutputter(string tempDir,
+                const vector<const pb::Protein*>& proteins,
+                VariableModTable* var_mod_table,
+                HeadedRecordWriter* final_writer,
+                unsigned long long memory_limit)
+    : tempDir_(tempDir),
+      modPeptideCnt_(0),
+      proteins_(proteins),
+      mod_table_(var_mod_table),
+      max_counts_(*mod_table_->MaxCounts()),
+      counts_mapper_vec_(max_counts_.size(), 0),
+      final_writer_(final_writer),
+      count_(0),
+      totalWritten_(0),
+      memory_limit_(memory_limit) {
+    numFiles_ = 1;
+    for (int i = 0; i < max_counts_.size(); ++i) {
+      counts_mapper_vec_[i] = numFiles_;
+      if (max_counts_[i] == 0)
+        numFiles_ *= (max_counts_[i]+2);
+      else
+        numFiles_ *= (max_counts_[i]+1);
     }
+    InitCountsMapper();
   }
 
-  ~ModsOutputterAlt() {
+  ~ModsOutputter() {
   }
 
-  void Output(pb::Peptide* peptide) {
-    if (maxMods_ < 0) {
-      return;
-    } else if (FLAGS_min_mods < 1) {
-      WritePeptide(peptide); // write unmodified peptide
-    }
-    if (maxMods_ == 0) {
-      return;
-    }
-
-    ResultMods resultMods(modTable_, modMaxCounts_, maxMods_, peptide, proteins_);
-    double unmodifiedMass = peptide->mass();
+  uint64_t Total() const { return totalWritten_; }
```

Comment:
Is there a reason you remove the following comment?

"// Return the total number of peptides written"

Reply:
No idea. I put it back.